src/BinDeps.jl

```diff
@@ -1,37 +1,32 @@
 module BinDeps
     importall Base
 
+    using Compat
+
     export @make_run, @build_steps, find_library, download_cmd, unpack_cmd,
             Choice, Choices, CCompile, FileDownloader, FileRule,
             ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
             autotools_install, CreateDirectory, MakeTargets, SystemLibInstall
 
-    if VERSION >= v"0.4.0-dev+3844"
-        import Base.Libdl: dlext, dlpath, RTLD_LAZY, DL_LOAD_PATH
-    else
-        const dlext = isdefined(Base.Sys, :shlib_ext) ? Base.Sys.shlib_ext : Base.Sys.dlext # Julia 0.2/0.3 compatibility
-        dlpath = Sys.dlpath
-    end
-    const shlib_ext = dlext # compatibility with older packages (e.g. ZMQ)
+    const dlext = @compat Libdl.dlext
+    const shlib_ext = @compat Libdl.dlext # compatibility with older packages (e.g. ZMQ)
 
     function find_library(pkg,libname,files)
         Base.warn_once("BinDeps.find_library is deprecated, use Base.find_library instead.")
         dl = C_NULL
         for filename in files
-            dl = dlopen_e(joinpath(Pkg.dir(),pkg,"deps","usr","lib",filename))
+            dl = @compat Libdl.dlopen_e(joinpath(Pkg.dir(),pkg,"deps","usr","lib",filename))
```

Comment:
I don't understand why you need `@compat` here.  The Compat package defines a `Libdl` module.